rest_framework-stubs/permissions.pyi

```diff
@@ -1,16 +1,17 @@
-from typing import Any, Dict, List, Protocol, Sequence, Type, Union
+from typing import Dict, List, Protocol, Sequence, Type, Union
 
 from django.db.models import Model, QuerySet
-from django.views import View
 from rest_framework.request import Request
+from rest_framework.views import APIView
 
 SAFE_METHODS: Sequence[str] = ("GET", "HEAD", "OPTIONS")
 
 class _SupportsHasPermission(Protocol):
-    def has_permission(self, request: Request, view: View) -> bool: ...
-    def has_object_permission(self, request: Request, view: View, obj: Any) -> bool: ...
+    def has_permission(self, request: Request, view: APIView) -> bool: ...
+    def has_object_permission(self, request: Request, view: APIView, obj: Model) -> bool: ...
```

Comment:
view may operate on any object not models only, so obj should be Any I think

Reply:
Thats true regarding views in general (non generic ones at least), but object permissions are model specific in DRF afaik.

Reply:
No, only DjangoObjectPermissions class and its children are model specific.

Reply:
ok, fixed